packages/world/test/Factories.t.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { Test, console } from "forge-std/Test.sol";
+import { World } from "../src/World.sol";
+import { Create2Factory } from "../src/factories/Create2Factory.sol";
+import { WorldFactory } from "../src/factories/WorldFactory.sol";
+import { IWorldFactory } from "../src/factories/IWorldFactory.sol";
+import { CoreModule } from "../src/modules/core/CoreModule.sol";
+
+contract Factories is Test {
+  event ContractDeployed(address addr, uint256 salt);
+  event WorldDeployed(address indexed newContract);
+  event HelloWorld();
+
+  function calculateAddress(
+    address deployingAddress,
+    bytes32 salt,
+    bytes memory bytecode
+  ) internal pure returns (address) {
+    bytes32 bytecodeHash = keccak256(bytecode);
+    bytes32 data = keccak256(abi.encodePacked(bytes1(0xff), deployingAddress, salt, bytecodeHash));
+    return address(uint160(uint256(data)));
+  }
+
+  function testCreate2Factory() public {
+    Create2Factory create2Factory = new Create2Factory();
+    // Encode constructor arguments for WorldFactory
+    bytes memory encodedArguments = abi.encode(new CoreModule());
+    bytes memory combinedBytes = abi.encodePacked(type(WorldFactory).creationCode, encodedArguments);
+    // Address we expect for deployed WorldFactory
+    address calculatedAddress = calculateAddress(address(create2Factory), bytes32(0), combinedBytes);
+    // Confirm event for deployment
+    vm.expectEmit(true, false, false, false);
+    emit ContractDeployed(calculatedAddress, uint256(0));
+    create2Factory.deployContract(combinedBytes, uint256(0));
+    // Confirm worldFactory was deployed correctly
+    IWorldFactory worldFactory = IWorldFactory(calculatedAddress);
+    assertEq(uint256(worldFactory.worldCount()), uint256(0));
+  }
+
+  function testWorldFactory() public {
+    // Deploy WorldFactory with current CoreModule
+    address worldFactoryAddress = address(new WorldFactory(new CoreModule()));
+    IWorldFactory worldFactory = IWorldFactory(worldFactoryAddress);
+    // Address we expect for World
+    address calculatedAddress = calculateAddress(worldFactoryAddress, bytes32(0), type(World).creationCode);
+    // Check for HelloWorld event from World
+    vm.expectEmit(true, false, false, false);
+    emit HelloWorld();
+    // Check for WorldDeployed event from Factory
+    vm.expectEmit(true, false, false, false);
```

Comment:
high level style nit - let's add an empty line before every comment to improve readability